lib/renderer/content-scripts-injector.ts

```diff
@@ -108,15 +108,17 @@ ipcRendererUtils.handle('CHROME_TABS_EXECUTE_SCRIPT', function (
   return runContentScript.call(window, extensionId, url, code)
 })
 
-module.exports = (getRenderProcessPreferences: typeof process.getRenderProcessPreferences) => {
-  // Read the renderer process preferences.
-  const preferences = getRenderProcessPreferences()
-  if (preferences) {
-    for (const pref of preferences) {
-      if (pref.contentScripts) {
-        for (const script of pref.contentScripts) {
-          injectContentScript(pref.extensionId, script)
-        }
+type ContentScriptEntry = {
+  extensionId: string;
+  contentScripts: Electron.ContentScript[];
+}
+
+module.exports = () => {
+  const entries = ipcRendererUtils.invokeSync<ContentScriptEntry[]>('CHROME_GET_CONTENT_SCRIPTS')
```

Comment:
Given @zcbenz 's comment I'm wondering how much of an impact this IPC can have, could you do a quick benchmark with `console.time` around this call to see how much slower the sync IPC call is compared to the previous voodoo.  Wouldn't block the PR but an interesting stat we should be aware of

Reply:
Especially considering this would run on _every_ page load 🤔

Reply:
It would add about 200ms load time: https://github.com/atom/atom/pull/1503.

Reply:
@nornagon If the 200ms number above is still accurate (2014 was a while ago) I'd prefer not to land this.  This is a hefty price to pay that we could simply avoid by _not_ doing this 🤔

Reply:
I timed it at about 1.6ms.

Reply:
(I think the PR referenced removes the crashreporter startup, which waits for a new process to start, which on Windows can take 100-200ms. This just returns a string.)